### PR TITLE
Add 2bit grayscale support

### PR DESCRIPTION
**Pull request by jedahan**
This lets people use framebuf for 2 color displays like the 8x8x2 matrix. It is a mostly mechanical port of the circuitpython GS2_HMSB code.

You can see how I use this code in https://github.com/jedahan/alarmclock , attached are terminal visualizations of `python3 blinkenlights.py` using some of this framebuf api:

Screenshot 2023-02-04 at 2 11 36 PM

Screenshot 2023-02-04 at 2 11 25 PM

https://user-images.githubusercontent.com/32407/216785487-a1d25784-6257-4a6b-b7d6-05d1a1f43dd0.mov

### CODE REVIEW

File: adafruit_framebuf.py
Comment (jedahan) [inline adafruit_framebuf.py:54]: I don't actually understand why we right shift 2, then left shift 1 and then left shift by `shift`. I wonder if there is a conceptually simpler way to write this.

**Comment by jedahan** [comment]
ping

**Comment by jposada202020** [comment]
Hello :). Thank you for your first PR in CircuitPython, as discussed in discord an example will help the test,
For new functionalities, it will be a good idea to add an example, this way people would use the new functionality.

**Review by FoamyGuy — APPROVED** [review]
This change looks good to me.

I tested successfully with the given "emulator" CPython script that outputs to the terminal and pared it down to a very basic simpletest-like script to validate the basic usage of framebuf, everything appears to be working as expected as far as I can tell.

I do not have the bicolor 8x8 matrix to test the new functionality on the actual hardware it was meant for, but don't think we need to hold this PR up for that reason.

I did regression testing with a 7x15 LED charliewing feather wing, a 16x8 LED Matrix featherwing, and a SSD1306 featherwing all of which utilized the frambuf repo (albeit not the new functionality) and confirmed all of those are functioning as expected. All testing was carred out on FEather RP2040 with version `8.0.3`.

Thanks for adding this new functionality @jedahan. Sorry that it took us a while to get feedback here. 

Going to merge this now, if someone gets hardware in the future to test it and finds problems we can create a new issue / PR as needed to resolve.

**Comment by jedahan** [comment]
I just noticed this got merged, thank you!

I quickly updated the terminal framebuffer example to be a little bit more minimal, would you want a PR for something like this? https://gist.github.com/jedahan/92adc7242e20376bbad64bcbf5ace72c

**Pull request merged**
